rpc/client.go

```diff
@@ -80,13 +82,16 @@ func NewHttpRpcClient(rpcServerUrl string) (*RpcClient, error) {
 	return c, nil
 }
 
-func (rc *RpcClient) GetVirtualChannel(id types.Destination) query.PaymentChannelInfo {
-	req := serde.GetPaymentChannelRequest{Id: id}
+func (rc *RpcClient) GetVirtualChannel(chId types.Destination) (query.PaymentChannelInfo, int, error) {
+	if (chId == types.Destination{}) {
+		return query.PaymentChannelInfo{}, http.StatusBadRequest, errors.New("a valid channel id must be provided")
+	}
+	req := serde.GetPaymentChannelRequest{Id: chId}
 
-	return waitForRequest[serde.GetPaymentChannelRequest, query.PaymentChannelInfo](rc, serde.GetPaymentChannelRequestMethod, req)
+	return waitForRequest[serde.GetPaymentChannelRequest, query.PaymentChannelInfo](rc, serde.GetPaymentChannelRequestMethod, req), http.StatusOK, nil
```

Comment:
I'm not sure about returning an http status code. Although in practice we tend to use http as a transport for the RPC connection, it is in fact transport agnostic.

If the input validation were being performed by the RPC server (instead of the client), it could make more sense to return an http error (again depending on the transport). 

In fact, thinking about this some more, I wonder if it does in fact make more sense to the validation on the server side.

The main thrust behind having validation is to avoid the nitro node panicking https://github.com/statechannels/boost/pull/3#discussion_r1223402151 . 

The deeper in we put the validation, the more protection we get. Putting the validation in the go rpc client means we don't get protection when
* someone uses the TS rpc client (unless we duplicate the validation there too)
* someone uses go-nitro as a library, not as a service

So I think we should rethink our approach a little. Perhaps start with a test which triggers the go-nitro node to panic by sending in a query for a zero channel using the rpc client. Then, we would like to transform that panic to an error, bubble it up to the rpc sever, and send it back to the rpc client as an json rpc error response. Ultimately that will be received by the rpc client and it will still cause an error to be returned from `RpcClient.GetVirtualChannel`.

Keen to hear your thoughts! And apologies that the direction wasn't clear from the outset.

Reply:
That makes sense to me and I agree that pushing validation deeper would be better. I noticed that in our jsonrpc server.go we have methods for `GetPaymentChannel` and `GetLedgerChannel`. Would `GetVirtualChannel` be the same as `GetPaymentChannel` or should I create an additional jsonrpc method?

Reply:
There are a few other instances in the rpc client code where "Virtual" is used instead of "Payment". To improve the naming consistency, would you agree with renaming the following client methods?

`CreateVirtual` --> `CreatePaymentChannel`
`CloseVirtual` --> `ClosePaymentChannel`
`CreateLedger` --> `CreateLedgerChannel`
`CloseLedger` --> `CloseLedgerChannel`

Reply:
I wonder if we might also want to align some of the json-rpc methods with the names of the core/node methods. The following method names are a little confusing:

json-rpc | node
-- | --
virtual_fund | CreateVirtualPaymentChannel
virtual_defund | CloseVirtualChannel
direct_fund | CreateLedgerChannel
direct_defund | CloseLedgerChannel